client/resource_group/controller/model.go

```diff
@@ -133,9 +144,15 @@ func newSQLCalculator(cfg *Config) *SQLCalculator {
 	return &SQLCalculator{Config: cfg}
 }
 
-// Trickle ...
-// TODO: calculate the SQL CPU cost and related resource consumption.
-func (dsc *SQLCalculator) Trickle(ctx context.Context, consumption *rmpb.Consumption) {
+func (dsc *SQLCalculator) SetConfig(config *Config) {
+	dsc.Config = config
+}
+
+// Trickle update sql layer CPU consumption.
+func (dsc *SQLCalculator) Trickle(consumption *rmpb.Consumption) {
+	delta := getSQLProcessCPUTime(dsc.isSingleGroupByKeyspace) - consumption.SqlLayerCpuTimeMs
+	consumption.TotalCpuTimeMs += delta
+	consumption.SqlLayerCpuTimeMs += delta
```

Comment:
we need make it start from 0.

Reply:
We can assign the sql time at the time of initialization?